# HG changeset patch
# User Frediano Ziglio <frediano.ziglio@citrix.com>
# Date 1336394267 -3600
# Node ID efb4061b9ce8b36fe64571ffecda732dacf4aab0
# Parent  a2ff1bd3a2005e6a16a83075d0fa78c2c6edc256
vgabios: Report mode not supported getting mode informations

If you try to get mode information for an unsupported mode
interrupt should return error but not that the function is not
supported.

Signed-off-by: Frediano Ziglio <frediano.ziglio@citrix.com>
Committed-by: Keir Fraser <keir@xen.org>

diff -r a2ff1bd3a200 -r efb4061b9ce8 tools/firmware/vgabios/vbe.c
--- a/tools/firmware/vgabios/vbe.c	Mon May 07 13:37:10 2012 +0100
+++ b/tools/firmware/vgabios/vbe.c	Mon May 07 13:37:47 2012 +0100
@@ -909,7 +909,8 @@ Bit16u *AX;Bit16u ES;Bit16u DI;
 void vbe_biosfn_return_mode_information(AX, CX, ES, DI)
 Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u DI;
 {
-        Bit16u            result=0x0100;
+        // error by default is 0x014f which means supported but error
+        Bit16u                 result=0x014f;
         Bit16u            ss=get_SS();
         ModeInfoBlock     info;
         ModeInfoListItem  *cur_info;
@@ -953,7 +954,6 @@ Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u D
 #ifdef DEBUG
                 printf("VBE *NOT* found mode %x\n",CX);
 #endif
-                result = 0x100;
         }
         
         if (result == 0x4f)
